lib/rspec/matchers/built_in/yield.rb

```diff
@@ -317,16 +329,16 @@ def expected_arg_description
         def negative_failure_reason
           if !@probe.has_block?
             'was not a block'
-          elsif all_args_match?
+          elsif @args_matched_when_yielded && !@expected.empty?
```

Comment:
I think this logic is the same at what was there previously (except this uses the yield-time values), but I'd appreciate a second opinion.